source/geod24/LocalRest.d

```diff
@@ -164,71 +173,325 @@ private struct ArgWrapper (T...)
+        this._belongToNode = value;
+    }
+
+    public @property bool belongToNode () @safe @nogc pure
+    {
+        return this._belongToNode;
+    }
+
+    /***************************************************************************
+
+        It is a function that accepts Message
+
+        Params:
+            msg = The message to send.
+
+    ***************************************************************************/
+
+    public void send (T) (T msg) @trusted
+    {
+        if (is (T == Message))
+            this.chan.send(cast(Message)msg);
+        else
+            this.chan.send(Message(msg));
+    }
+
+
+    /***************************************************************************
+
+        Return the received message.
+
+        Returns:
+            A received `Message`
+
+    ***************************************************************************/
+
+    public bool receive (Message *msg) @trusted
+    {
+        return this.chan.receive(msg);
+    }
+
+
+    /***************************************************************************
+
+        Return the received message.
+
+        Params:
+            msg = The `Message` pointer to receive.
+
+        Returns:
+            Returns true when message has been received. Otherwise false
+
+    ***************************************************************************/
+
+    public bool tryReceive (Message *msg) @trusted
+    {
+        return this.chan.tryReceive(msg);
+    }
+
+
+    /***************************************************************************
+
+        Close the `Channel`
+
+    ***************************************************************************/
+
+    override public void close () @trusted
+    {
+        this.chan.close();
+    }
+
+
+    /***************************************************************************
+
+        Return closing status
+
+        Return:
+            true if channel is closed, otherwise false
+
+    ***************************************************************************/
+
+    override public @property bool isClosed () @safe @nogc pure
+    {
+        return false;
+    }
+
+
+    /***************************************************************************
+
+        Generate a convenient string for identifying this LocalTransceiver.
+
+    ***************************************************************************/
+
+    override public void toString (scope void delegate(const(char)[]) sink)
+    {
+        import std.format : formattedWrite;
+        formattedWrite(sink, "TR(%x)", cast(void*) chan);
+    }
+}
+
+/***************************************************************************
+
+    Getter of LocalTransceiver assigned to a called thread.
+
+    Returns:
+        Returns instance of `LocalTransceiver` that is created by top thread.
+
+***************************************************************************/
+
+private @property LocalTransceiver thisLocalTransceiver () nothrow
```

Comment:
This should be removed

Reply:
I've been thinking about this.
How to save information on running threads.
Traditionally, global variables have solved this.
This time, I created a new thread class. What do you think about this?
```
public class InfoThread : Thread
{
    public ThreadInfo info;
....
}
```